src/Services/MailSender.php

```diff
@@ -29,17 +30,21 @@ public function __construct(
     public function send(Email $email, ?Envelope $envelope = null): void
     {
         // let first send the mail itself, to see if it was really sent or entered error state
-        $this->mailSender->send($email, $envelope);
-
-        $this->saveMail($email);
+        try {
+            $this->mailSender->send($email, $envelope);
+            $this->saveMail($email);
+        }catch(MailTransportFailedException $e){
+            $this->saveMail($email, true);
+            throw $e;
+        }
```

Comment:
```suggestion
       $failed = false;

        try {
            $this->mailSender->send($email, $envelope);
        } catch(MailTransportFailedException $e) {
            $failed = true;
            throw $e;
        } finally {
            $this->saveMail($email, $failed);
        }
```

What are your thoughts on using `finally` instead?

Reply:
That is what I did first. Then I thought introducing a new variable just to keep track of whether the delivery failed or not seems a bit unnecessary. However I'm completely fine with your solution. Should I commit your suggestion?

Reply:
Reading the code without IDE gave me: what? Why parameter? Oh, that's why.

In my opinion, it is clearer to read.